test/functional/core/fileio_spec.lua

```diff
@@ -64,5 +66,25 @@ describe('fileio', function()
     command('write')
     eq(4, request('nvim__stats').fsync)
   end)
+
+  it('creates a backup', function()
+    clear({ args={ '-i', 'Xtest_startup_shada',
+                   '--cmd', 'set directory=Xtest_startup_swapdir' } })
+
+    command('write Xtest_startup_file1')
+    feed('ifoo<esc>')
+    command('set backup')
+    command('set backupcopy=yes')
+    command('write')
+    feed('Abar<esc>')
+    command('write')
+
+    local foo_contents = read_file('Xtest_startup_file1')
+    local bar_contents = read_file('Xtest_startup_file1~')
+
+    eq('foobar\n', foo_contents);
+    eq('foo\n', bar_contents);
```

Comment:
on Windows lines end with `\r\n`. To make this test cross-platform, could just trim whitespace. 

There is a trim() function in `test/unit/helpers.lua`: 

https://github.com/neovim/neovim/blob/230a4fc2c327a24bd9515304c9197f8fb91313c7/test/unit/helpers.lua#L123                                                   

We need to lift that up into `test/helpers.lua` so that functional tests can use it.

Reply:
I forgot about that. I moved the trim function to `test/helpers.lua` and used it in my test.